frontend/src/features/admin-form/responses/ResponsesPage.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import { useParams } from 'react-router'
+import { Skeleton } from '@chakra-ui/react'
+
+import Button from '~components/Button'
+
+import { useAdminFormSettings } from '../settings/queries'
+
+import { downloadEncryptedResponses } from './AdminSubmissionsService'
+import { useFormResponses } from './queries'
+
+const ResponsesPage = (): JSX.Element => {
+  const { data: settings } = useAdminFormSettings()
+  const { data, isLoading } = useFormResponses()
+  const { formId } = useParams()
+  const [secretKey, setSecretKey] = useState<string>('')
+
+  const handleCsvExport = async () => {
+    await downloadEncryptedResponses(formId!, settings!.title, secretKey)
```

Comment:
does this not block the web ui?

Reply:
Oh yes my bad, there shouldn't be an await here!